src/datachain/lib/video.py

```diff
@@ -0,0 +1,185 @@
+import posixpath
+import shutil
+import tempfile
+from typing import Optional
+
+from numpy import ndarray
+
+from datachain.lib.file import (
+    FileError,
+    ImageFile,
+    Video,
+    VideoFile,
+    VideoFragment,
+    VideoFrame,
+)
+
+try:
+    import ffmpeg
+    import imageio.v3 as iio
+except ImportError as exc:
+    raise ImportError(
+        "Missing dependencies for processing video.\n"
+        "To install run:\n\n"
+        "  pip install 'datachain[video]'\n"
+    ) from exc
+
+
+def video_info(file: VideoFile) -> Video:
+    """
+    Returns video file information.
+
+    Args:
+        file (VideoFile): Video file object.
+
+    Returns:
+        Video: Video file information.
+    """
+    try:
+        probe = ffmpeg.probe(file.get_local_path())
+    except ffmpeg.Error as exc:
+        raise FileError(file, f"unable to probe video file: {exc.stderr}") from exc
+    except Exception as exc:
+        raise FileError(file, f"unable to probe video file: {exc}") from exc
+
+    if not probe:
```

Comment:
[C] Could collapse these checks by giving probe a default (`or {}`) above

```python
probe = ffmpeg.probe(file.get_local_path()) or {}
...
all_streams = probe.get("streams")
video_format = probe.get("format")
if not all_streams or not video_format:
        raise FileError(file, "unable to probe video file")
```

Reply:
and as the error is user-facing we might want to dumb it down a bit: "unable to extract metadata from video file" or something like that